stm32/tim.c

```diff
@@ -15,6 +15,37 @@
 static volatile int64_t timeoff;
 static volatile cb_t timer_cb;
 
+/*
+ * tim_delay_us - Busy-wait delay for a specified number of microseconds.
+ *
+ * This function implements a busy-wait loop that delays execution for at least
+ * the given number of microseconds. It relies on a 64-bit microsecond timer value,
+ * which is obtained from tim_get_micros(). Since the 64-bit timer value is built
+ * from a 16-bit hardware counter extended by an offset and may be updated by an
+ * interrupt (making its access non-atomic on a 32-bit MCU), the function disables
+ * interrupts while reading the timer value to prevent it from being updated during
+ * the read operation. This ensures a consistent, atomic snapshot of the 64-bit
+ * time value.
```

Comment:
@GJSea have seen this happen? the code in `tim_get_micros()` is meant to prevent that - it does two reads of the counter with a check of overflow flag in the middle. The `timeoff` is only updated on wake up with RTC and when the counter overflows.